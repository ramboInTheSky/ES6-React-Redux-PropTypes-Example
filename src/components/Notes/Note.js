import React from 'react';
import PropTypes from 'prop-types';
import { formatting } from 'nhh-styles';

import { NoteItem, NoteBody, NoteSubject, NoteDate, NoteAuthor } from './components';

export const NotePropType = {
  created: PropTypes.shape({
    on: PropTypes.string.isRequired,
    by: PropTypes.string,
  }).isRequired,
  id: PropTypes.string.isRequired,
  subject: PropTypes.string,
  text: PropTypes.string,
};

export const Note = ({ id, created, text, subject }) => (
  <NoteItem key={id} data-bdd={`note-item-${id}`}>
    <NoteBody>
      <NoteDate data-bdd={`note-${id}-date`}>
        {formatting.formatDate(created.on, formatting.dateTimeFormat)}
      </NoteDate>
      {created.by ? (
        <NoteAuthor data-bdd={`note-${id}-name`}>
          {' - '}
          {created.by}
        </NoteAuthor>
      ) : null}
      {subject ? <NoteSubject data-bdd={`note-${id}-subject`}>{subject}</NoteSubject> : null}
      {text && (
        <div>
          <span data-bdd={`note-${id}-text`}>{text}</span>
        </div>
      )}
    </NoteBody>
  </NoteItem>
);

Note.propTypes = NotePropType;

Note.defaultProps = {
  subject: '',
  text: '',
};
