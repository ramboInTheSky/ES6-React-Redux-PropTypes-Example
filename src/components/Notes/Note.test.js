import React from 'react';
import { shallow } from 'enzyme';
import * as nhhStyles from 'nhh-styles';
import { Note } from './Note';

nhhStyles.formatting.formatDate = jest.fn(() => '10/08/2018 - 8:52pm');

describe('<Note />', () => {
  let props;

  beforeEach(() => {
    props = {
      note: {
        id: 'e2054c02-08dd-e811-80d2-005056825b41',
        caseId: '00000000-0000-0000-0000-000000000000',
        created: { by: null, on: '2018-10-31T12:25:13+00:00' },
        modified: { by: null, on: '2018-10-31T12:25:13+00:00' },
        subject: 'usman-moses',
        text: 'This is a test note with attachment :)',
      },
    };
  });

  it('renders correctly', () => {
    expect(shallow(<Note {...props.note} />)).toMatchSnapshot();
  });
});
