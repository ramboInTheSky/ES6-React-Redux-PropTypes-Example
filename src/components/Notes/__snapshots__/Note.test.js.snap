// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Note /> renders correctly 1`] = `
<styled.div
  data-bdd="note-item-e2054c02-08dd-e811-80d2-005056825b41"
  key="e2054c02-08dd-e811-80d2-005056825b41"
>
  <styled.div>
    <styled.span
      data-bdd="note-e2054c02-08dd-e811-80d2-005056825b41-date"
    >
      10/08/2018 - 8:52pm
    </styled.span>
    <styled.span
      data-bdd="note-e2054c02-08dd-e811-80d2-005056825b41-subject"
    >
      usman-moses
    </styled.span>
    <div>
      <span
        data-bdd="note-e2054c02-08dd-e811-80d2-005056825b41-text"
      >
        This is a test note with attachment :)
      </span>
    </div>
  </styled.div>
</styled.div>
`;
